File: arrays-and-strings/3334-apple-redistribution-into-boxes/apple-redistribution-into-boxes.java
class Solution {
    public int minimumBoxes(int[] apple, int[] capacity) {
        
        Arrays.sort(capacity);
        int total=0;
        for(int i=0;i<(int)apple.length;i++){
            total+=apple[i];
        }
        int n=capacity.length;
        for(int i=n-1;i>=0;i--){
            total-=capacity[i];
            if(total<=0)return n-i;
        }
        return -1;
    }
}

File: arrays-and-strings/3334-apple-redistribution-into-boxes/apple-redistribution-into-boxes.cpp
class Solution {
public:
    int minimumBoxes(vector<int>& apple, vector<int>& capacity) {
        sort(capacity.begin(),capacity.end());
        
    
        int total=0;
        for(int i=0;i<(int)apple.size();i++){
            total+=apple[i];
            
        }
        int n=capacity.size();
        for(int j=n-1;j>=0;j--){
            total-=capacity[j];
            if(total<=0){
                return (n-j);
            }
        }
        return -1;
    }
};